src/_js/main.mjs

```diff
@@ -13,25 +13,55 @@
 
 import '/_js/dark-mode-toggle.mjs';
 
-// Dark mode toggle.
 const darkModeToggle = document.querySelector('dark-mode-toggle');
-const root = document.documentElement;
 
+// Toggles the `dark` class based on the dark mode toggle's mode
+const root = document.documentElement;
 const updateThemeClass = () => {
   root.classList.toggle('dark', darkModeToggle.mode === 'dark');
 };
 
+// Only load the Twitter script when we need it.
+const twitterLink = document.querySelector('.twitter-link');
+let twitterLoaded = null;
+if (twitterLink) {
+  twitterLoaded = import('https://platform.twitter.com/widgets.js')
+    .then(() => twitterLink.remove());
+}
+
+// Dynamically either insert the dark- or the light-themed Twitter widget.
+let twitterTimelineContainer = document.querySelector('.twitter-widget');
+const updateTwitterTimeline = async () => {
+  if (twitterTimelineContainer) {
+    await twitterLoaded;
+    const newContainer = twitterTimelineContainer.cloneNode();
```

Comment:
This container should probably be hidden initially, otherwise I guess UI would jump around?

Reply:
Oh, right, it still doesn't resolve if it's hidden because iframe onload is not firing...

Reply:
Good point, fixed.

Reply:
It works when the container is not showing, so we're good I guess.